File: dcp-solutions/2025-01-03/solution.go
```markdown
// Definition for a binary tree node.
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isSubtree(s *TreeNode, t *TreeNode) bool {
	if s == nil || t == nil {
		return false
	}

	if solve(s, t) {
		return true
	}

	return isSubtree(s.Left, t) || isSubtree(s.Right, t)
}

func solve(root1, root2 *TreeNode) bool {
	if root1 == nil && root2 == nil {
		return true
	}
	if root1 == nil || root2 == nil {
		return false
	}
	if root1.Val != root2.Val {
		return false
	}

	return solve(root1.Left, root2.Left) && solve(root1.Right, root2.Right)
}
```

File: dcp-solutions/2025-01-03/solution.c
```c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 *     struct TreeNode *root; // extra pointer to self 
 * };
 */
bool isSubtree(struct TreeNode* s, struct TreeNode* t) {
    return check_subtree(s, t) || isSubtree(s->left, t) || isSubtree(s->right, t);
}

bool check_subtree(struct TreeNode* s, struct TreeNode* t) {
    if (!s && !t) return true;
    if (!s || !t) return false;

    // check if the subtree t exists within the tree s
    bool result = (check_node_match(s, t)) ? true : false;
    return result && check_subtree(s->left, t) && check_subtree(s->right, t);
}

bool check_node_match(struct TreeNode* s, struct TreeNode* t) {
    if (!s && !t) return true;
    if (!s || !t) return false;

    // Check for matching values
    if (s->val != t->val) return false;

    // Recursively match left and right subtrees
    return check_node_match(s->left, t->left) &&
           check_node_match(s->right, t->right);
}
```